### ISSUE TITLE
scala-xml-1 strips comments on load

### ISSUE THREAD

**Issue by armanbilge** [http4s/http4s-scala-xml#32]
```
==> X org.http4s.scalaxml.ScalaXmlSuite.round trips utf-83.458s munit.FailException: Failing seed: DnQ9OBcjAHkCfLgo7An2adTPVELxRySHaRbsQsgfceD=
You can reproduce this failure by adding the following override to your suite:
  override def scalaCheckInitialSeed = "DnQ9OBcjAHkCfLgo7An2adTPVELxRySHaRbsQsgfceD="
Exception raised on property evaluation.
> ARG_0: <红麛鶣蓕塘羪><!--媀돊ߕ--></红麛鶣蓕塘羪>
> Exception: munit.ComparisonFailException: /home/runner/work/http4s-scala-xml/http4s-scala-xml/scala-xml/src/test/scala/org/http4s/scalaxml/ScalaXmlSuite.scala:66
65:        .as[Elem]
66:        .assertEquals(elem)
67:    }
values are not the same
=> Obtained
<红麛鶣蓕塘羪/>
=> Diff (- obtained, + expected)
-<红麛鶣蓕塘羪/>
+<红麛鶣蓕塘羪><!--媀돊ߕ--></红麛鶣蓕塘羪>
```

**Comment by armanbilge** [comment]
Actually not sure if this was a flake.

**Comment by rossabaker** [comment]
This looks like a comment serialization issue.

**Comment by rossabaker** [comment]
I think it's https://github.com/scala/scala-xml/commit/f92ae49181989764d591f5b2b6d843ce9d12bfa8.  If we fix that, it will change the behavior of scala-xml-1.  If we don't, we'll need to have two separate normalizing rules for the tests.

**Comment by armanbilge** [comment]
I'm actually not sure what needs to change where 😅 I've organized it in https://github.com/http4s/http4s-scala-xml/pull/29 so it should be easy to add separate sources/tests for each module, I think that's easier than a switch?

**Comment by rossabaker** [comment]
Yes, that already does the nasty build work.  I appended a commit to #29 that should handle this discrepancy.

**Comment by rossabaker** [comment]
Fixed in #29
